### PR TITLE
Look for all mounts before removing a volume

### PR DESCRIPTION
**Pull request by ulyssessouza**
**What I did**
This checks the volume mounts of all container groups from all
resource groups before removing a volume/fileshare

**Related issue**
https://github.com/docker/compose-cli/issues/733

**Tests**
/test-aci

**(not mandatory) A picture of a cute animal, if possible in relation with what you did**

### CODE REVIEW

File: aci/volumes.go [hunk 1]
```diff
@@ -164,7 +165,47 @@ func errorEvent(resource string) progress.Event {
...
+		if v.AzureFile != nil && v.AzureFile.StorageAccountName != nil && v.AzureFile.ShareName != nil &&
+			(*v.AzureFile.StorageAccountName+"/"+*v.AzureFile.ShareName) == id {
+			return true
+		}
+	}
+	return false
+}
+
 func (cs *aciVolumeService) Delete(ctx context.Context, id string, options interface{}) error {
+	err := checkVolumeUsage(ctx, cs.aciContext, id)
```
Comment (ulyssessouza) [inline aci/volumes.go:205]: Maybe we should add a `--force` option to avoid the check...

File: aci/volumes.go [hunk 2]
```diff
@@ -164,7 +165,47 @@ func errorEvent(resource string) progress.Event {
 	}
 }
 
+func checkVolumeUsage(ctx context.Context, aciContext store.AciContext, id string) error {
+	resourceGroups, err := NewACIResourceGroupHelper().ListGroups(ctx, aciContext.SubscriptionID)
+	if err != nil {
+		return err
+	}
+	for _, rg := range resourceGroups {
```
Comment (gtardif) [inline aci/volumes.go:173]: You can probably avoid this loop, and use an ACI call that does not filter on resource group to get containers

Comment (ulyssessouza) [reply aci/volumes.go:173]: I tried this already. The problem is that from the list of `container groups` I cannot know from which `resource group` it comes from.

File: aci/volumes.go (hunk 1 above)
Comment (gtardif) [reply aci/volumes.go:205]: maybe... not a blocker for this PR, we can add this later on if we want. cc @nebuk89

**Review by gtardif — APPROVED** [review]

**Comment by gtardif** [comment]
rebased

**Pull request merged**